sinatra-contrib/lib/sinatra/capture.rb

```diff
@@ -96,8 +96,7 @@ def capture(*args, &block)
         result = block[*args]
       elsif current_engine == :erb || current_engine == :slim
         @_out_buf, _buf_was = '', @_out_buf
-        block[*args]
-        result = eval('@_out_buf', block.binding)
+        result = block.call(*args)
```

Comment:
You should try to get result from `@_out_buf` unless returned value is empty.
```ruby
raw = block.call(*args)
captured = block.binding.eval('@_out_buf')
result = captured.empty? ? raw : captured
```

See: https://github.com/padrino/padrino-framework/blob/02feacb6afa9bce20c1fb360df4dfd4057899cfc/padrino-helpers/lib/padrino-helpers/output_helpers/abstract_handler.rb#L36

Reply:
Good call 👍